hydra/test_utils/launcher_common_tests.py

```diff
@@ -494,23 +516,30 @@ def test_custom_sweeper_run_workdir(
         task_launcher_cfg: DictConfig,
         extra_flags: List[str],
     ) -> None:
-        overrides = extra_flags + overrides
+        overrides = extra_flags + overrides + self.custom_overrides()
         task_launcher_cfg = OmegaConf.create(task_launcher_cfg or {})  # type: ignore
         task_config = OmegaConf.create(task_config or {})  # type: ignore
         cfg = OmegaConf.merge(task_launcher_cfg, task_config)
         assert isinstance(cfg, DictConfig)
+        working_dir = self.working_dir()
+        if working_dir:
+            expected_outputs = str(working_dir / expected_dir)
+        else:
+            expected_outputs = str(tmpdir / expected_dir)
```

Comment:
This is another example.

You are sprinkling logic that is designed to work around challanges in the ray launcher plugin in too many places.
We need to minimize these things to the Ray launcher itself and it's testing code.
Let's talk about it today.

Food for thought:
Can the download logic be a part of the launcher plugin itself that can be activated via a config flag in the test?